src/Analysis/Ast/Impl/Types/PythonClassType.Generics.cs

```diff
@@ -259,33 +256,40 @@ internal partial class PythonClassType {
         /// Points the generic type parameter in class type to their corresponding specific type (or a generic
         /// type parameter if no specific type was provided)
         /// </summary>
-        private void StoreGenericParameters(PythonClassType classType, IGenericTypeParameter[] genericParameters, IReadOnlyDictionary<IGenericTypeParameter, IPythonType> genericToSpecificTypes) {
+        private void StoreGenericParameters(
+            IReadOnlyDictionary<IGenericTypeParameter, IPythonType> currentGenericParameters,
+            IEnumerable<IGenericTypeParameter> newGenericParameters,
+            IReadOnlyDictionary<IGenericTypeParameter, IPythonType> genericToSpecificTypes) {
+
             // copy original generic parameters over and try to fill them in
-            classType._genericParameters = new Dictionary<IGenericTypeParameter, IPythonType>(GenericParameters.ToDictionary(k => k.Key, k => k.Value));
+            _genericParameters = currentGenericParameters.ToDictionary(k => k.Key, k => k.Value);
 
             // Case when creating a new specific class type
             if (Parameters.Count == 0) {
```

Comment:
This is also now a different `Parameters`. Maybe these two functions should just stay as they were since they access so much of the original class...

Reply:
It needs to receive template class from which it will only access public members.

Reply:
`SetClassMembers(IPythonClassType templateClass, ...)` and `void StoreGenericParameters(
            IPythonClassType templateClass, ...`